### PR TITLE
fix: wildcards on Windows (Issue: #1521)

### LINKED ISSUE #1521: Unexpected error when specifying a wildcarded path for linting on Windows
**Describe the bug**

When trying to run `@redocly/cli lint` and providing a wildcarded path on Windows the command exist with `exit code 1` without any error text.

**To Reproduce**

1. Make sure you are on Windows. I used v11, and Node 20, but it's also reproducible on Node v18. Haven't tried other Nodes. 
2. Create an empty Node.js project by running `npm init` in a directory of your choice
3. Add `@redocly/cli` to your project by running `npm install @redocly/cli`
4. Create a very simple project structure:
```
<ROOT>
├─ api
│  └─ providers
│     └─ test-provider
│        └─ api.yaml (put "hello world" in this file, I know it's invalid YAML, but it doesn't matter)
``` 
5. Run `npx @redocly/cli lint api/providers/*/api.yaml` in the root folder

**Expected behavior**

I would expect to see the following text with an error on screen:
```
No configurations were provided -- using built in recommended configuration by default.

validating api/providers/test-provider/api.yaml...
Something went wrong when processing api/providers/test-provider/api.yaml:

  - Document must be JSON object, got string.
```

**Actual behavior**

Nothing is displayed. Command exits with exit code 1.

**`Node.js` Version(s)**

20

**Additional context**

I managed to find the cause of this issue. When `lint` is called we get the following sequence of calls:

```TypeScript
handleLint               // redocly-cli/packages/cli/src/commands/lint.ts:      line 40
getgetFallbackApisOrExit // redocly-cli/packages/cli/src/commands/lint.ts:      line 41
expandGlobsInEntrypoints // redocly-cli/packages/cli/src/utils/misceleneous.ts: line 51
```
And in the `expandGlobsInEntrypoints` we have this [line](https://github.com/Redocly/redocly-cli/blob/90a79eeab90f43fbf60cead3c7ba0450905f4394/packages/cli/src/utils/miscellaneous.ts#L109C10-L109C98): 
```TypeScript
 ? (await glob.__promisify__(aliasOrPath)).map((g: string) => getAliasOrPath(config, g))
```

You can see the call to `__promisify__` function which does not exist on `glob`. I don't really know the reason why `@types/glob` added this, but the function does not exist and the correct way to promisify this would be:

```TypeScript
// redocly-cli/packages/cli/src/utils/misceleneous.ts
import { promisify } from 'utils';

// expandGlobsInEntrypoints
/* line 109 */ await promisify(glob)(aliasOrPath).map(...)
```

### PR DESCRIPTION
**Pull request by dmitri-trofimov**
## What/Why/How?
This PR fixes and issue with handling wildcards in command-line on Windows (#1521).

## Testing
Tested locally on my machine.

## Check yourself

- [x] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by lornajane** [comment]
This looks great, thanks @dmitri-trofimov

File: .changeset/soft-berries-remain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@redocly/cli": patch
+---
+
+Fix handling of wildcards on Windows ([#1521](https://github.com/Redocly/redocly-cli/issues/1521)).
```
Comment (tatomyr) [inline .changeset/soft-berries-remain.md:5]:
```suggestion
Fixed handling of wildcards on Windows ([#1521](https://github.com/Redocly/redocly-cli/issues/1521)).
```

**Review by tatomyr — APPROVED** [review]
LGTM. The changes will be available in the next release.

**Comment by dmitri-trofimov** [comment]
Thank you so much!

**Pull request merged**